File: client-app/src/app/models/university.ts
import { City, RegionDto } from "./region";

export interface University {
	id: string;
	name: string;
	accreditation: Accreditation;
	region: RegionDto;
	city: City;
	address: string;
	rating: number;
	website: string;
	info: string;
	telephone: string;
	ukraineTop: number;
	studentsCount: number;
	titlePhoto: string;
	location: Location;
	inFavoriteList: boolean;
}

export interface Accreditation {
	id: number;
	accreditationLevel: string;
}

export interface Location {
	id: number;
	longitude: number;
	latitude: number;
}

export class University implements University {
	constructor(init?: UniversityFormValues) {
		Object.assign(this, init);
	}
}

export class UniversityFormValues {
	id: string = '';
	name: string = '';
	accreditation?: Accreditation;
	region?: RegionDto;
	city?: City;
	address: string = '';
	rating: number = 0;
	website: string = '';
	info: string = '';
	telephone: string = '';
	ukraineTop: number = 0;
	studentsCount: number = 0;
	titlePhoto: string = '';
	location?: Location;

	constructor(university?: UniversityFormValues) {
		if (university) {
			this.id = university.id;
			this.name = university.name;
			this.region = university.region;
			this.city = university.city;
			this.address = university.address;
			this.website = university.website;
			this.info = university.info;
			this.studentsCount = university.studentsCount;
			this.titlePhoto = university.titlePhoto;
		}
	}
}

export interface UniversityQueryParams {
	name: string,
	accreditationId: number,
	regionsId: number[],
	citiesId: number[],
	degreeId: number,
	knowledgeBranchesId: string[],
	specialtyBasesId: string[],
	minPrice: number,
	maxPrice: number,
	budget: boolean,
	ukraineTop: boolean
}

File: Domain/University.cs
namespace Domain
{
	public class University
	{
		public Guid Id { get; set; }
		public string Name { get; set; }
		public Region Region { get; set; }
		public City City { get; set; }
		public Location Location { get; set; }
		public Accreditation Accreditation { get; set; }
		public string Address { get; set; }
		public string Website { get; set; }
		public string Info { get; set; }
		public string Telephone { get; set; }
		public int UkraineTop { get; set; }
		public Photo TitlePhoto { get; set; }
		public ICollection<Photo> Photos { get; set; } = new List<Photo>();
		public ICollection<Faculty> Faculties { get; set; } = new List<Faculty>();
		public ICollection<Review> Reviews { get; set; } = new List<Review>();
		public ICollection<UniversityAdmin> UniversityAdmins { get; set; } = new List<UniversityAdmin>();
		public ICollection<FavoriteList> FavoriteList { get; set; } = new List<FavoriteList>();
	}
}